esp/services/ws_rrd/RRDGraphWrapper.cpp

```diff
@@ -0,0 +1,287 @@
+                {
+                    strRRDFilePath.append(pServer);
+                }
+                else
+                {
+                    strRRDFilePath.append(pSummaryInfo);
+                }
+                strRRDFilePath.append("/");
+            }
+            strRRDFilePath.appendf("%s.rrd", pRRDFile);
+
+            if (checkFileExists(strRRDFilePath.str()) == false)
+            {
+                PROGLOG("Can not find rrd file for '%s'", strRRDFilePath.str());
+                throw MakeStringException(0, "Can not find rrd file for '%s'", strRRDFilePath.str());
+            }
+
+            strRRDGraphCmd.appendf(" %s%d%d=%s", pDataDefinitionPrefix1, idxMetric+1, idx+1, strRRDFilePath.str());
+            strRRDGraphCmd.appendf("%s", pDataDefinitionSuffix);
+            strRRDGraphCmd.appendf(" %s%d:ds%d%d#%s:%s", pLineDefinition, nLineCount++, idxMetric+1, idx+1, PlotColor[(nLineCount-1)%nPlotColors], pRRDFile);
+
+            if (bServerSpecified == false || bClusterSpecified == false)
+            {
+                break;
+            }
+        }
+    }
+
+    strRRDGraphCmd.appendf(" %s %d", pWidth, nWidth);
+    strRRDGraphCmd.appendf(" %s %d", pHeight, nHeight);
+
+    if (pTitle != NULL && *pTitle != 0)
+    {
+        strRRDGraphCmd.appendf(" -t '%s'", pTitle);
+    }
+
+    PROGLOG("RRDTOOL GRAPH CMD --> %s <--", strRRDGraphCmd.str());
+    FILE *fp = popen(strRRDGraphCmd.str(), "r");
+
+    if (fp == NULL)
+    {
+        return NULL;
+    }
+
+    pclose(fp);
+
+    fp = fopen(strDestinationPath.str(),"r");
+
+    if (fp == 0)
+    {
+        return NULL;
+    }
+
+    CFileInputStream svgFileIStream(fileno(fp));
+
+    int nCharacter = -1;
+
+    pBuffer->clear();
+
+    StringBuffer strBufferIn;
+
+    char removedChars[64] = {0};
+
+    int nCount = 0;
+    do
+    {
+        nCharacter = svgFileIStream.readNext();
+
+        if (nCharacter != -1 && (nCount > 37 || bGenPNG == true))  // remove xml tag in svg
+        {
+            strBufferIn.append(static_cast<unsigned char>(nCharacter));
+        }
+        else if (nCharacter != -1)
+        {
+            removedChars[nCount] = nCharacter;
+        }
+        nCount++;
+    }
+    while(nCharacter != -1);
+
+    if (*removedChars != 0)
+    {
+        DBGLOG("Removing XML tag:%s", removedChars);
+    }
+
+    StringBuffer strBufferIn2(strBufferIn.str());
+    strBufferIn.clear().set(strBufferIn2.str());
+
+    fclose(fp);
+
+    const int nOffSet = 4;
+    char originalID[1024] = "";
+    char *pBufferIn = new char[strBufferIn.length()+1];
+    strncpy(pBufferIn, strBufferIn.str(), strBufferIn.length()+1);
+
+    char *pStart = NULL;
+    char *pEnd = NULL;
+
+    pStart = strstr(const_cast<char*>(pBufferIn), "id=\"");
+    pEnd = strstr(pStart+nOffSet, "\"");
+
+    while (pStart != NULL)
+    {
+        int len = pEnd-pStart-nOffSet;
+
+        if (len > 1024)
+        {
+            PROGLOG("SVG id too long");
```

Comment:
The ID is short.  Looks like a variant of "glyph0-0".  1024 is more than enough for all IDs that I have encountered.